src/util/analytics-client.ts

```diff
@@ -0,0 +1,91 @@
+import {getLogger} from "../config/logger";
+import { omit } from "lodash";
+import { optionalRequire } from "optional-require";
+import {AnalyticsConfigEnum} from "../interfaces/common";
+import {isNodeProd} from "./isNodeEnv";
+
+const { analyticsClient } = optionalRequire("@atlassiansox/analytics-node-client") || {};
+const logger = getLogger("analytics")
+const instance = process.env.INSTANCE_NAME;
+const appKey = `com.github.integration${instance ? `.${instance}` : ""}`;
+
+function sendAnalytics(eventType: "trait")
+function sendAnalytics(eventType: "screen", attributes:{name:string} & Record<string,unknown>)
+function sendAnalytics(eventType: "ui" | "track" | "operational", attributes: Record<string,unknown>)
+function sendAnalytics(eventType: string, attributes?: Record<string, unknown>): void {
+	if(!analyticsClient && !isNodeProd()){
+		return;
+	}
+
+	const analyticsNodeClient = analyticsClient({
+		env: AnalyticsConfigEnum.ProdEnv,
+		product: AnalyticsConfigEnum.Product
+	});
```

Comment:
You really should need to create your analytics client on every `sendAnalytics` call.  You should have a check to see if it's already created and have a variable outside the function, like 
```
let analyticsNodeClient:AnalyticsClient;

function sendAnalytics(eventType: string, attributes?: Record<string, unknown>): void {
	if(!analyticsClient && !isNodeProd()){
		return;
	}

  if(!analyticsNodeClient) {
	  analyticsNodeClient = analyticsClient({
		  env: envVars.MICROS_ENV,
		  product: "gitHubForJira"
	  });
  }
...
```

Reply:
That wouldn't work as `!analyticsNodeClient` would've returned the function...